lib/cisco_node_utils/vxlan_vtep.rb

```diff
@@ -90,121 +90,48 @@ def description=(desc)
-    #      param_hash = { name: @name, no_cmd: 'no', vni: diff[1],
-    #                     mcast_grp: diff[2] }
-    #      config_set('vxlan_vtep', 'member_vni_mgrp', param_hash)
-    #      param_hash = { name: @name, no_cmd: '', vni: diff[1],
-    #                     mcast_grp: diff[3] }
-    #    end
-    #    config_set('vxlan_vtep', 'member_vni_mgrp', param_hash)
-    #  end
-    # rescue CliError => e
-    #  raise "[vxlan_vtep #{@name}] '#{e.command}' : #{e.clierror}"
-    # end
-
-    def mac_distribution
-      mac_dist = config_get('vxlan_vtep', 'mac_distribution', name: @name)
-      if mac_dist.nil?
-        @mac_dist_proto = :flood
-      else
-        @mac_dist_proto = (mac_dist == 'bgp') ? :evpn : :flood
-      end
-      @mac_dist_proto.to_s
-    end
-
-    def mac_distribution=(val)
-      if val == :flood
-        if @mac_dist_proto == :evpn
-          config_set('vxlan_vtep', 'mac_distribution',
-                     name: @name, state: 'no', proto: 'bgp')
-        end
-        @mac_dist_proto = :flood
-      elsif val == :evpn
-        config_set('vxlan_vtep', 'mac_distribution',
-                   name: @name, state: '', proto: 'bgp')
-        @mac_dist_proto = :evpn
+    def host_reachability
+      hr = config_get('vxlan_vtep', 'host_reachability', name: @name)
+      hr == 'bgp' ? 'evpn' : hr
+    end
+
+    def host_reachability=(val)
+      set_args = { name: @name, proto: 'bgp' }
+      if val.to_s == 'flood' && host_reachability == 'evpn'
+        set_args[:state] = 'no'
+      elsif val.to_s == 'evpn'
+        set_args[:state] = ''
       end
-    rescue Cisco::CliError => e
-      raise "[#{@name}] '#{e.command}' : #{e.clierror}"
+      config_set('vxlan_vtep', 'host_reachability', set_args)
+    end
+
+    def default_host_reachability
+      config_get_default('vxlan_vtep', 'host_reachability')
     end
 
     def source_interface
-      src_intf = config_get('vxlan_vtep', 'source_intf', name: @name)
-      return default_source_interface if src_intf.nil?
-      src_intf
+      config_get('vxlan_vtep', 'source_intf', name: @name)
+    end
+
+    def source_interface_set(val)
+      set_args = { name: @name, lpbk_intf: val }
+      val.empty? ? set_args[:state] = 'no' : set_args[:state] = ''
+      config_set('vxlan_vtep', 'source_intf', set_args)
     end
 
     def source_interface=(val)
-      fail TypeError unless val.is_a?(String)
-      if val.empty?
-        config_set('vxlan_vtep', 'source_intf',
-                   name: @name, state: 'no', lpbk_intf: val)
-      else
-        config_set('vxlan_vtep', 'source_intf',
-                   name: @name, state: '', lpbk_intf: val)
-      end
-    rescue Cisco::CliError => e
-      raise "[#{@name}] '#{e.command}' : #{e.clierror}"
+      # The source interface can only be changed if the nve
+      # interface is in a shutdown state.
+      current_state = shutdown
+      self.shutdown = true unless shutdown
+      source_interface_set(val)
```

Comment:
Muddies the waters too much for my taste.  IMO it's easier to understand the shut/noshut logic without the extra unrelated code.